library/windows/glfw.gni

```diff
@@ -0,0 +1,22 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```

Comment:
The variables in this file are all only used in the one windows BUILD.gn file, right? If so, they can just be in that file instead of broken out. Same for jsoncpp.gni.

The reason engine.gni exists is that I needed some values in two different sections of the build system, rather than an organizational system. When possible, values should stay local to the file than needs them.

Reply:
I'm not convinced `build/win/config/BUILD.gn` should exist, actually, vs. those being in the file that declares the targets.

The thing about the linux configs is that they are properties of the system, not of the project's build setup, so they aren't associated with any library target. These Windows configs, on the other hand, do have concrete targets (albeit not typical ones) that they belong to, and that something wanting the config should be depending on. Given that, it makes sense for them to live with the target.

That said, I don't think it's a big deal either way, and I expect to rearrange parts of the GN build as it evolves (macOS support, non-GLFW implementations) so I'm fine proceeding with the structure as-is, and coming back to this question later.